lib/WebDriver/AbstractWebDriver.php

```diff
@@ -122,15 +124,30 @@ protected function curl($requestMethod, $command, $parameters = null, $extraOpti
 
         list($rawResult, $info) = ServiceFactory::getInstance()->getService('service.curl')->execute($requestMethod, $url, $parameters, $extraOptions);
 
+        // According to the spec:
+        // - valid responses always include: sessionId, status and value
+        //   (at least as long as 'they include a message body' - but we tested commands such as timeouts/async_script
+        //   and session/delete and they still give back a json payload)
+        // - failed commands (http 500) include those as well
+        // - invalid requests (http 4xx) are instead plaintext
+        // See https://w3c.github.io/webdriver/webdriver-spec.html for details
+
+        if ($info['http_code'] >= 400 && $info['http_code'] <= 499) {
```

Comment:
- can do < 500
- I did not rely on 1st digit only because I was unsure about status codes > 999 being valid, but at least according to rfc7231 they are not, so can do